Add SessionWindow details page with simple analysis

janpeterka opened the pull request:
To better understand data from `rrweb`, I started building simple `session_window/details` page.
It then spawned some base work on SessionWindow analysis, which could be first step in exploration of
> Searching / filtering of recordings, including navigation and 404s/500s, button clicks, rage clicks, dead clicks, etc.

which I would love to look into :grin: 

This PR adds:
- some basic parsing of event data (types, source, href)
- little analysis from this data (list of visited pages)
- page for viewing session window data

It's pretty rough, but hope it might be good first step.

![Screenshot from 2024-12-29 19-32-39](https://github.com/user-attachments/assets/1e9c2955-db18-415d-9659-65eec8903ce7)
![Screenshot from 2024-12-29 19-32-45](https://github.com/user-attachments/assets/79d618d7-cd6e-4c2b-89fa-6f357cba30cd)

janpeterka commented on app/views/spectator_sport/dashboard/session_windows/details.html.erb:
```diff
@@ -0,0 +1,26 @@
+<%= link_to "Back to dashboard", "/spectator_sport_dashboard" %>
```
Ugh, this again. Will fix soon based on your changes.

bensheldon replied:
I just merged my PR

janpeterka commented on app/models/spectator_sport/session_window_analysis.rb:
```diff
@@ -0,0 +1,17 @@
+
+    def tags # simple naive implementation of tagging sessions
+      tags = []
+      tags << "server_error" if visited_pages.any? { _1.include?("500.html") }
```
> In future, error page could probably emit custom event for simpler detection. I looked into custom events a bit, but so far wasn't able to use them.

janpeterka replied:
> The 500.html error page is rarely served directly.
Yes, you are right that this doesn't really make sense :sweat_smile:
 
> will want to create a custom controller/acton that raises an exception
I'm not sure what you mean by that. You mean for the demo app app?

bensheldon replied:
> You mean for the demo app app?

yes, for the demo app 😅

janpeterka replied:
I originally did that, but it does show dev error page instead of 500.html (as it's run in dev enviroment by default)
I found out we could force that using `config.consider_all_requests_local = false`, maybe for demo its sort of ok default behavior?

The question remains then - how to detect that the page shown is error.
Other than some middleware i can only think of rrweb custom event as a general solution.
Maybe you have some other ideas?

I can remove the error labeling from this PR until we solve that, unless we find some straightforward solution.

janpeterka replied:
Agree, will remove error tagging for now, and will let you know when PR will be ready for another look :)

Thanks a lot!

bensheldon replied:
Got it figured out: https://github.com/bensheldon/spectator_sport/pull/17

janpeterka replied:
Sorry, I'm super busy at work at the moment, so I didn't get to this.

janpeterka commented:
Ok, finally I got back to this
- removed error tracking
- updated to current code + design

I left there now empty `tags`, which I'm not sure if I should've, can remove it. I still hope I will soon look into how this data can be analysed and tagged usefully :grin:
